docs/testing/developer-smoke-test-cases.md

```diff
@@ -53,6 +54,25 @@ We currently do **not** provide support for or test against the following enviro
 + Clusters with 1 or more Windows workers interacting with **multiple** domains at once, either joined or un-joined
 + Clusters interacting with **multiple** Active Directory instances at once
 
+## Create a `ClusterRoleBinding` for your service account
+
+In order to test the Account Provider you must first install the GMSA Web-hook, which will expand the contents of `GMSACredentialSpec` resources onto pods. This chart will deploy a purpose built `ClusterRole` which permits access to all `GMSACredentialSpec` resources. However, the default `ClusterRoleBinding` deployed by the GMSA Web-hook chart only grants access to these resources to the service account used by the GMSA web-hook. In order to create workloads which utilize a `GMSACredentialSpec`, but run under a different service account, an additional `ClusterRoleBinding` needs to be created. 
+
+```yaml 
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: <CLUSTER_ROLE_BINDING_NAME>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rancher-windows-gmsa # This is automatically created when the GMSA Web-hook chart is installed 
+subjects:
+  - kind: ServiceAccount
+    name: <SERVICE_ACCOUNT_NAME>
+    namespace: <SERVICE_ACCOUNT_NAMESPACE>
+```
```

Comment:
I don't like the approach of using the ClusterRole we added into the GMSA Webhook deployment that was intended to only be used by the GMSA Webhook Deployment.

We should tell a developer to create the Role, RoleBinding, and GMSACredentialSpec all in one shot along with their sample workload instead of just binding their workload to the one that allows their workload to use all specs, since that more accurately mirrors what users would do.